packages/collection/src/collection.ts

```diff
@@ -11,11 +11,11 @@ export type ReadonlyCollection<Key, Value> = Omit<
 
 export interface Collection<Key, Value> {
 	/**
-	 * Ambient declaration to allow `this.constructor[@@species]` in class methods.
+	 * Ambient declaration to allow references to `this.constructor` in class methods.
 	 *
 	 * @internal
 	 */
-	constructor: typeof Collection & { readonly [Symbol.species]: typeof Collection };
+	constructor: typeof Collection;
 }
```

Comment:
In-class `declare` syntax specifically balks at a property named `constructor`, so this declaration has to remain in its own interface rather than being merged into the class body.